src/InterventionBackend.php

```diff
@@ -835,4 +844,14 @@ protected function isStreamReadable($stream)
         $meta = stream_get_meta_data($stream);
         return isset($meta['mode']) && (strstr($meta['mode'], 'r') || strstr($meta['mode'], '+'));
     }
+
+    /**
+     * This gets the temp path config usually set through local_temp_path
+     *
+     * @return string
+     */
+    public function getTempPathConfig(): string
+    {
+        return $this->config()->get('local_temp_path') ?? TEMP_PATH;
```

Comment:
The fallback doesn't seem strictly necessary here, since the default config value is TEMP_PATH already. I'd also suggest casting the return from config as a string, since it can return null as well (e.g. if you haven't flushed yet but pull this code in)

Reply:
Does `private static $local_temp_path = TEMP_PATH;` definitely work as expected? It’s a const we define ourselves in framework, so it strikes me as potentially fragile... I think my preference would be to remove it from the config definition and keep it here

Reply:
(Not in SS3 but yes in SS4)

Reply:
Would you not want it to throw an exception if config returns null rather than trying to create a file in a random path?

Reply:
The problem I'm trying to highlight is that it can be null while you're flushing, if that method is called during the flush process

Reply:
Fair enough, have updated it to cast to a string 👍

Reply:
@adrhumphreys could you also address @kinglozzer's comment? You'll need to rebase and fix the conflict as well =)